testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MainActivity.java

```diff
@@ -246,46 +236,65 @@ public void onAcquireTokenClicked(final AcquireTokenFragment.RequestOptions requ
         callAcquireToken(mScopes, mUiBehavior, mLoginHint, mExtraQp, mExtraScopesToConsent);
     }
 
-    public void onRemoveUserClicked(String username) {
-        final List<IAccount> accountsToRemove = mApplication.getAccounts();
-
-        for (final IAccount accountToRemove : accountsToRemove) {
-            if (TextUtils.isEmpty(username) || accountToRemove.getUsername().equals(username.trim().toLowerCase())) {
-                    mApplication.removeAccount(accountToRemove);
+    public void onRemoveUserClicked(final String username) {
+        mApplication.getAccounts(new PublicClientApplication.AccountsLoadedListener() {
+            @Override
+            public void onAccountsLoaded(List<IAccount> accountsToRemove) {
+                for (final IAccount accountToRemove : accountsToRemove) {
+                    if (TextUtils.isEmpty(username) || accountToRemove.getUsername().equals(username.trim().toLowerCase())) {
```

Comment:
how about using equalsIgnoreCase and removing the toLower, that way we get independent of the casing of acccountToRemove